snuba/datasets/cdc/cdcprocessors.py

```diff
@@ -116,11 +112,6 @@ def _process_delete(
     def process_message(self, value, metadata) -> Optional[ProcessedMessage]:
         assert isinstance(value, dict)
 
-        partition = metadata.partition
-        assert (
-            partition == KAFKA_ONLY_PARTITION
-        ), "CDC can only work with single partition topics for consistency"
-
```

Comment:
This takes a step in making the MessageProcessor fully Kafka agnostic. Now we still have the offset lingering around in the KafkaMetadata, but that's the last thing to remove.